crates/router/src/connector/dlocal/transformers.rs

```
@@ -0,0 +1,527 @@
+    pub installments_id: Option<String>,
+    pub installments: Option<String>,
+}
+
+#[derive(Debug, Default, Eq, PartialEq, Serialize)]
+pub struct ThreeDSecureReqData {
+    pub force: bool,
+}
+
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct DlocalPaymentsRequest {
+    pub amount: i64, //amount in cents, hence passed as integer
+    pub currency: enums::Currency,
+    pub country: Option<String>,
+    pub payment_method_id: String,
+    pub payment_method_flow: String,
+    pub payer: Payer,
+    pub card: Option<Card>,
+    pub order_id: String,
+    pub notification_url: String,
+    pub three_dsecure: Option<ThreeDSecureReqData>,
+    pub callback_url: Option<String>,
+    // pub wallet: Option<String>,
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for DlocalPaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        match item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => {
+                let should_capture = matches!(
+                    item.request.capture_method,
+                    Some(enums::CaptureMethod::Automatic) | None
+                );
+                let payment_request = Self {
+                    amount: item.request.amount,
+                    country: Some(get_currency(item.request.currency)),
+                    currency : item.request.currency,
+                    payment_method_id :  "CARD".to_string(),
+                    payment_method_flow : "DIRECT".to_string(),
+                    payer : Payer {
+                        name: ccard.card_holder_name.peek().clone(),
+                        email: match &item.request.email{
+                            Some (c) => c.peek().clone().to_string(),
+                            None => "dummyEmail@gmail.com".to_string()
```

If email is mandatory, please add a validation here and throw Missing field error incase email is not present.

---

Added validation